codeforpdx/PASS
# Add checksum to ttl documents + a little refactoring

Changes:
- added a few new npm scripts:
  - `dev:pod:stored` lets developers start a solid server in their own filesystem.
  - `test` runs the testing framework `vitest`
  - `coverage` gives a unit test coverage report. It checks the codepaths in tested files to see if all possible paths are hit.
- Added `UPLOAD_TYPES` and `RDF_PREDICATES` constants to a new constants directory. It's usually best to store all project constants in one place.
- added `crypto-js` as a project dependency
- added `vitest`, a unit testing library, as a dev dependency + a few other modules vitest depends on
- added a checksum to the document.ttl files that links the ttl to a specific resource. The checksum is generated by sha256 and should be very difficult to fake. Currently, the RDF is using the `serialNumber` predicate from schema.org, but there's probably a better one somewhere.

Testing:
- Upload the same .pdf file twice, once as a bank statement, once as a passport.
- Upload a different .pdf file as a drivers license
- Check the document.ttl files generated by all three. The `serialNumber` value for the bank statement and the passport should be the same, but the serialNumber for the drivers license should be different.
